### ISSUE TITLE
ActiveSupport::TimeWithZone#inspect should include nsec in the output string

### ISSUE THREAD

**Issue by Fire-Dragon-DoL** [rails/rails#41971]
### Steps to reproduce

```ruby
t1 = Time.use_zone("UTC") { Time.zone.parse("2021-04-14T19:15:58.788642111Z") }
t2 = Time.use_zone("UTC") { Time.zone.parse("2021-04-14T19:15:58.788642000Z") }

t1 == t2 # false
t1.inspect # "Wed, 14 Apr 2021 19:15:58 UTC +00:00"
t2.inspect # "Wed, 14 Apr 2021 19:15:58 UTC +00:00"
t1.iso8601 # "2021-04-14T19:15:58Z"
t2.iso8601 # "2021-04-14T19:15:58Z"
t1.iso8601(3) # "2021-04-14T19:15:58.788Z"
t2.iso8601(3) # "2021-04-14T19:15:58.788Z"
t1.iso8601(6) # "2021-04-14T19:15:58.788642Z"
t2.iso8601(6) # "2021-04-14T19:15:58.788642Z"
```

It's only at `9` that we start seeing something. While this is not directly a bug, it would be tremendously helpful to include the nanoseconds directly in the object's `inspect` method, which is often used for debugging

### Expected behavior

```ruby
t1 = Time.use_zone("UTC") { Time.zone.parse("2021-04-14T19:15:58.788642111Z") }
t2 = Time.use_zone("UTC") { Time.zone.parse("2021-04-14T19:15:58.788642000Z") }
t1.inspect # "Wed, 14 Apr 2021 19:15:58.788642111 UTC +00:00"
t2.inspect # "Wed, 14 Apr 2021 19:15:58.788642000 UTC +00:00"
```

### Actual behavior

Nanoseconds are hidden entirely from the string

### System configuration
**Rails version**: 6.0.3.6
**Ruby version**: 2.7.2

**Comment by eugeneius** [comment]
This was implemented in https://github.com/rails/rails/pull/38874 and released in Rails 6.1.

**Comment by Fire-Dragon-DoL** [comment]
Amazing! Thank you
